connector/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/Serializer.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.ml
+
+import org.apache.spark.connect.proto
+import org.apache.spark.ml.linalg.{Matrix, Vector}
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.connect.common.LiteralValueProtoConverter
+
+object Serializer {
+
+  def serialize(data: Any, objIdentifier: String): proto.MlCommandResponse = {
+    data match {
+      case v: Vector => serializeVector(v)
+      case v: Matrix => serializeMatrix(v)
+      case _: Byte | _: Short | _: Int | _: Long | _: Float | _: Double | _: Boolean | _: String |
+           _: Array[_] =>
+        proto.MlCommandResponse
+          .newBuilder()
+          .setLiteral(LiteralValueProtoConverter.toLiteralProto(data))
+          .build()
+
+      case _: Dataset[_] =>
+        proto.MlCommandResponse.newBuilder()
+          .setIsDataframe(true)
+          .build()
+
+      case _ => // if didn't match, we just return the method chain to client
```

Comment:
What would reach here?

Reply:
We must have the default match for scala.

Reply:
This default logic is error-prone.

Reply:
My bad, this is needed for current implementation. Eg, for model.summary, server will return "model_id.summary" to client side.